source/http2_stream_manager.c

```diff
@@ -390,11 +394,119 @@ static void s_aws_http2_stream_manager_build_transaction_synced(struct aws_http2
     s_sm_log_stats_synced(stream_manager);
 }
 
+static void s_on_ping_complete(
+    struct aws_http_connection *http2_connection,
+    uint64_t round_trip_time_ns,
+    int error_code,
+    void *user_data) {
+
+    (void)http2_connection;
+    struct aws_h2_sm_connection *sm_connection = user_data;
+    if (error_code) {
+        return;
+    }
+    if (!sm_connection->connection) {
+        return;
+    }
+    AWS_ASSERT(aws_channel_thread_is_callers_thread(aws_http_connection_get_channel(sm_connection->connection)));
+    if (round_trip_time_ns > sm_connection->stream_manager->connection_ping_timeout_ns) {
+        /* Timeout happened. It may be caught by the timeout task first, but still close the connection. */
+        STREAM_MANAGER_LOGF(
+            ERROR,
+            sm_connection->stream_manager,
+            "ping timeout detected for connection: %p from ping_complete, round trip time in ns is: %" PRIu64
+            ". Closing connection.",
+            (void *)sm_connection->connection,
+            round_trip_time_ns);
+        aws_http_connection_close(sm_connection->connection);
```

Comment:
debatable: why bother closing here? we're already doing this from the timeout task? seems like an unnecessary bit of complexity to do the same thing from two places. If we fix a bug in one place we will probably not remember to fix it in the other

Reply:
yeah, will remove this. Doesn't do anything actually